### PR TITLE
Email & Pre-detection of postits

### PR DESCRIPTION
**Pull request by GeoffroyBERNARD**
+ A very simple mail containing the file link is sent if the email field is set (wasn't really planned for the sprint but was quick to do)

+ The camera button now activates a new component (Camera.vue) using trackingjs to get the camera stream and live-detect post-its. --There's no options to specify the camera we're using so we're stuck on the front cam for now.

+ Clicking on the video takes a snapshot that can then be uploaded the usual way.

+ About opening drawIo directly. It's possible but only once it has been opened manually at least once. I can't think of a feature that would make use of it with the way it works

### CODE REVIEW

File: src/components/Camera.vue [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            this.displayVideo = true;
+            let video = document.getElementById('video');
+            let canvas = document.getElementById('canvas');
+            let context = canvas.getContext('2d');
+            let tracker = new tracking.ColorTracker(['magenta', 'cyan', 'yellow']);
+            tracking.track('#video', tracker, { camera: true });
+            tracker.on('track', function(event) {
+                context.clearRect(0, 0, canvas.width, canvas.height);
+                event.data.forEach(function(rect) {
+                if (rect.color === 'custom') {
```
Comment (floo51) [inline src/components/Camera.vue]: Bad indentation

File: src/components/Camera.vue [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+        //use tracking js to display the camera and the post it in live
+        takePhoto() {
+            this.displayVideo = true;
+            let video = document.getElementById('video');
+            let canvas = document.getElementById('canvas');
+            let context = canvas.getContext('2d');
+            let tracker = new tracking.ColorTracker(['magenta', 'cyan', 'yellow']);
+            tracking.track('#video', tracker, { camera: true });
+            tracker.on('track', function(event) {
```
Comment (floo51) [inline src/components/Camera.vue]: This can probably be an arrow function

File: src/components/Camera.vue [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            let canvas = document.createElement('canvas');
+            let ctx = canvas.getContext('2d');
+            let video = document.getElementById('video');
+            canvas.width = video.width;
+            canvas.height = video.height
+            ctx.drawImage(video, 0, 0);
+            canvas.toBlob((blob) => {
+                let img = document.createElement('img');
+                img.src = window.URL.createObjectURL(blob);
+                this.$emit('screenshot', img);
```
Comment (floo51) [inline src/components/Camera.vue]: Poorly named event, would you say "screenshot" is an event?

I suggest `screenshotTaken`.

File: src/components/Postiterie.vue [hunk 4]
```diff
@@ -238,26 +243,51 @@ export default {
...
     // TODO
     // no IO
     // call the mail API or our mail server
     prepareMail: function(){
-        console.log("prepareMail was fired");
+        //  Using smtpjs to send mail
+        // TODO : smtpjs can encrypt credentiel and return a token but we need to pass a domain name so come back here for deplyment
+        Email.send({
+          Host : "smtp.gmail.com",
+          Username : process.env.VUE_APP_SMTP_EMAIL,
```
Comment (floo51) [inline src/components/Postiterie.vue]: You probably don't want SMTP email and password to appear in clear in the compiled output in everyone's browser.

File: src/components/Postiterie.vue [hunk 5]
```diff
@@ -238,26 +243,51 @@ export default {
...
+          Password : process.env.VUE_APP_SMTP_PASSWORD,
+          To : this.email,
+          From : process.env.VUE_APP_SMTP_EMAIL,
+          Subject : "Nouveau schéma généré",
+          Body : "Voici un lien vers le schéma que vous venez de générer " + this.link
+      }).catch(e => {
+        console.log(e);
```
Comment (floo51) [inline src/components/Postiterie.vue]: Display an actual error to the user?

File: src/components/TakePhoto.vue [hunk 6]
```diff
@@ -1,17 +1,22 @@
 <template>
-    <div class="big-button-container">
+
+    <div class="big-button-container" v-on:click="clicked">
         <label for="photo" class="button-top big-button"><i class="material-icons">photo_camera</i></label>
         <label for="photo" class="button big-button">Prendre une photo</label>
-        <input style="display:none" id="photo" type="file" @change="fileUpload($event.target.name, $event.target.files)"  accept="image/*;capture=camera">
+        <!--<input style="display:none" id="photo" type="file" @change="fileUpload($event.target.name, $event.target.files)"  accept="image/*;capture=camera">-->
```
Comment (floo51) [inline src/components/TakePhoto.vue]: Commented-out code shouldn't be merged on master

File: src/components/Postiterie.vue (hunk 4 above)
Comment (GeoffroyBERNARD) [reply src/components/Postiterie.vue]: SmtpJS can generate a security token that will allow us to remove our credentials from the script. 
I've set it up this way for now as the feature requires a domain name

**Pull request merged**